ktor-retrofit/src/main/kotlin/com/bnorm/ktor/retrofit/RetrofitService.kt

```diff
@@ -170,6 +182,20 @@ private fun Route.process(service: Any, function: KFunction<*>) {
   TODO("implement the rest of the function annotations : $annotations")
 }
 
+private fun checkRegisteredRoute(
+  method: HttpMethod,
+  baseUrl: Route,
+  path: String,
+  methodName: String
+) {
+  val newRoute = RegisteredRoute(method, baseUrl.toString(), path, methodName)
+  val registeredPaths = registeredRoutes.getOrDefault(method, mutableSetOf())
```

Comment:
Define `registeredRoutes` as `MutableMap<HttpMethod, MutableSet<RegisteredRoute>>` and then use `getOrPut` here instead. Cuts down on a number of conversions and interactions with the map.

Reply:
Fixed